eos/scattering/ee-to-ccbar_TEST.cc

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2021 Méril Reboud
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <test/test.hh>
+#include <eos/maths/complex.hh>
+#include <eos/scattering/ee-to-ccbar.hh>
+#include <eos/utils/kmatrix-impl.hh>
+
+#include <array>
+#include <memory>
+
+
+using namespace test;
+using namespace eos;
+
+class eetoccbarTest :
+public TestCase
+{
+public:
+    eetoccbarTest() :
+    TestCase("ee->ccbar scattering tests")
+    {
+    }
+
+    virtual void run() const
+    {
+
+        constexpr double eps = 1e-5;
+
+        Parameters p = Parameters::Defaults();
+        p["ee->ccbar::g0(psi(2S),ee)"] = 0.1;
+        p["ee->ccbar::g0(psi(3770),ee)"] = 0.2;
+        p["ee->ccbar::g0(psi(2S),D^0Dbar^0)"] = 0.3;
+        p["ee->ccbar::g0(psi(3770),D^0Dbar^0)"] = 0.4;
+        p["ee->ccbar::g0(psi(2S),D^+D^-)"] = 0.5;
+        p["ee->ccbar::g0(psi(3770),D^+D^-)"] = 0.6;
+
+
+        //Set all cst to zero
+        p["ee->ccbar::c_0(ee,ee)"] = 0.0;
+        p["ee->ccbar::c_0(ee,D^0Dbar^0)"] = 0.0;
+        p["ee->ccbar::c_0(ee,D^+D^-)"] = 0.0;
+        p["ee->ccbar::c_0(D^0Dbar^0,D^0Dbar^0)"] = 0.0;
+        p["ee->ccbar::c_0(D^0Dbar^0,D^+D^-)"] = 0.0;
+        p["ee->ccbar::c_0(D^+D^-,D^+D^-)"] = 0.0;
+
+        // Build K Matrix
+        auto psi2S_res = std::make_shared<charmonium_resonance<3, 2, 0>>("psi2S_res", p["mass::psi(2S)"], p["size::psi(2S)"]);
```

Comment:
You put the suffix ``_res`` in here. Similarly for the channels (``_chan``). Can we dispense with that?
Whenever a channel's or resonance's name is printed, the class should make clear what it is.